scipy-1.0/minimize.tex

```diff
@@ -0,0 +1,118 @@
+direction. Trust-region (TR) methods, on the other hand, fixes the maximum step length for which the local approximation
+is valid and finds approximately the best point inside this \textit{trust-region}.
+
+An example of trust-region method introduced in the latest releases is \texttt{trust-exact}, which a  provides a fast convergence rate by solving 
+ almost exactly the trust-region subproblem. Its downside is it also require the Hessian
+to be stored and factored every iteration, which may renders impossible the solution of
+very large problems.
+
+Some available unconstrained methods are appropriate for large-scale problems
+(i.e. problems with more than 10000 variables).
+\texttt{CG} method does not use second derivatives and is adequate to large-scale problems due to the low storage requirements.
+\texttt{Newton-CG}, \texttt{trust-ncg} and \texttt{trust-krylov} use the second order derivatives in order to get faster
+convergence but are still appropriate for large-scale problems because they factorate the Hessian in a inexact and iterative way,
+which does not require the Hessian to be actually stored.
+
+The method \texttt{trust-krylov}, introduced in the latest release, presents itself as a good compromise between
+\texttt{trust-ncg} and \texttt{trust-exact}: it provides an more accurate solution to the trust-region
+subproblem than \texttt{trust-ncg} but does not require the Hessian to be stored  and factored as in \texttt{trust-exact}.
+
+Besides unconstrained problems, the \texttt{minimize} function also provides methods for solving
+minimization problems subject to constraints. Table~\ref{tab:minimize-constrained} summarizes all
+constrained optimization methods available, with information of which type of
+constraints they are able to deal with.
+
+
+
+
+
+\begin{table}
+  \centering
+  \begin{tabular}{ccccc}
+     &  \rotatebox{60}{\texttt{L-BFGS-B}} & \rotatebox{60}{\texttt{TNC}} & \rotatebox{60}{\texttt{COBYLA}} & \rotatebox{60}{\texttt{SLSQP}}\\
+    \hline
+    Version added & $<$0.7 & $<$0.7 & $<$0.7 & 0.9 \\
+    \hline
+    It is a wrapper? & \cmark & \cmark & \cmark & \cmark\\
+    \hline
+    \nth{1} derivatives & \cmark & \cmark &  & \cmark \\
+    \hline
+    \nth{2} derivatives & Approx. & Approx. &  & Approx. \\
+    \hline
+    Line-search (LS) or trust-region (TR) & LS & LS & TR & LS \\
+    \hline
+    Bound constraint: $x^l \le x \le x^u$ & \cmark & \cmark &  & \cmark \\
+    \hline
+    Equality constraint: $c_{\text{eq}}(x) = 0$ &  &  &  & \cmark \\
+    \hline
+    Inequality constraint: $c_{\text{ineq}}(x) \ge 0$ &  &  & \cmark & \cmark \\
+    \hline
+    References & \cite{byrd_limited_1995, zhu_algorithm_1997}  & \cite{nash_newton-type_1984} &
+     \cite{powell_direct_1994, powell_direct_1998, powell_view_2007} & \cite{kraft_software_1988} \\
+    \hline
+  \end{tabular}
+  \caption{Optimization methods from \texttt{minimize} for constrained minimization.}
```

Comment:
I would add a note about "COBYLA", that bound constraints $x^l \le x \le x^u$ can be expressed as inequality constraints $x \ge x^l$,  $-x \ge -x^u$ and equality constraints $c_{\text{eq}}(x) = 0$ can be expressed as $c_{\text{eq}}(x) \ge 0$ and $-c_{\text{eq}}(x) \ge 0$.
I don't understand why this isn't done automatically by `minimize` personally. Is it horrendously bad for the problem structure - so bad that it's not worth supporting (with a warning in the documentation)?

Reply:
Ok. I decide to proceed as follows:
1. I will say that COBYLA can deal with bound constraints. The user just have to do this trick you just said.
2. I still say it cannot deal with equality constraints. Since, expressing equality constraints as $c_{\text{eq}}(x) \ge 0$ and $-c_{\text{eq}}(x) \ge 0$ usually does not work well in practice.